### DISCUSSION TITLE
Support all OCI image fields

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12324] [Ideas]
## Problem

Some OCI image fields can be set by a Dockerfike but are not supported by the Dagger API. For example, `healthcheck`.

## Solution

Add missing functions to the `Container` type to reach feature parity with Dockerfiles.

**Comment by helderco** [comment]
We currently support all OCI `[ImageConfig](https://github.com/opencontainers/image-spec/blob/036563a4a268d7c08b51a08f05a02a0fe74c7268/specs-go/v1/config.go#L23-L50)` fields, but `healthcheck` is part of an [extended DockerOCI](https://github.com/moby/docker-image-spec/blob/f1d00ebd2d6d6805170d5543dbca4b850f35f9af/specs-go/v1/image.go#L27-L32) which adds Docker specific fields:

* `Healthcheck`
* `OnBuild`
* `Shell`

I suppose `OnBuild` doesn't make sense when using Dagger, but could be if publishing a base image that's intended to be used in a `Dockerfile` later. Do people still use that?

`Shell` is used to set the shell-form of `RUN`, `CMD` and `ENTRYPOINT`.

**Comment by alexcb** [comment]
#11911 adds support for `Healthcheck`

**Comment by alexcb** [comment]
To gauge demand for `OnBuild`, i.e. implementing a `container.OnBuild([]string{"/some/command/here"})`, please give this comment an upvote with a 👍

**Comment by alexcb** [comment]
To gauge demand for `Shell`, i.e. implementing a `container.Shell([]string{"/usr/local/bin/xonsh"})`, please give this comment an upvote with a 👍
